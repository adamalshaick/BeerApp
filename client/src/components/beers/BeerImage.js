import React from "react";
import { Image, ImageWrapper, Thumbnail } from "../common/styles/Image";
import ReactCSSTransitionGroup from "react-addons-css-transition-group";
import PropTypes from "prop-types";

const BeerImage = ({ showImage, beer, onClick }) => {
  const transitionOptions = { transitionName: "fade" };
  let image;
  if (showImage) {
    image = (
      <div key={beer}>
        <ImageWrapper onClick={onClick} />
        <Image
          src={beer.image_url}
          onError={e => {
            e.target.onerror = null;
            e.target.src =
              "https://www.equiinzert.cz/gallery/inzeraty/no-image.png";
          }}
          showImage={showImage}
        />
      </div>
    );
  }
  return (
    <>
      <Thumbnail
        showImage={showImage}
        src={beer.image_url}
        onError={e => {
          e.target.onerror = null;
          e.target.src =
            "https://www.equiinzert.cz/gallery/inzeraty/no-image.png";
        }}
        onClick={onClick}
      />

      <ReactCSSTransitionGroup {...transitionOptions}>
        {image}
      </ReactCSSTransitionGroup>
    </>
  );
};

BeerImage.propTypes = {
  showImage: PropTypes.bool.isRequired,
  beer: PropTypes.object.isRequired,
  onClick: PropTypes.func.isRequired
};

export default BeerImage;
