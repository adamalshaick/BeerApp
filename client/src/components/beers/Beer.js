import React, { Component } from "react";
import { BeersColumn } from "../common/styles/StyledColumn";
import PropTypes from "prop-types";
import BeerImage from "./BeerImage";

class Beer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showImage: false
    };
  }
  // onClick = e => {
  //   e.preventDefault();
  //   this.setState({ showImage: true });
  // };

  // onClose = e => {
  //   e.preventDefault();
  //   this.setState({ showImage: false });
  // };

  render() {
    const { beer } = this.props;
    return (
      <BeersColumn className="row">
        <div className="col-8">
          <strong>name:</strong> {beer.name}
          <br />
          <strong>type:</strong> {beer.type}
          <br />
          <strong>price:</strong> {beer.price}
        </div>
        <div className="col-4">
          <BeerImage
            showImage={this.state.showImage}
            beer={beer}
            onClick={() => {
              this.setState(prevState => ({
                showImage: !prevState.showImage
              }));
            }}
          />
        </div>
      </BeersColumn>
    );
  }
}

Beer.propTypes = {
  beer: PropTypes.object.isRequired
};

export default Beer;
