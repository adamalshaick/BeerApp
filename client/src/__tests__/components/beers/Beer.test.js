import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Beer from "../../../components/beers/Beer";
import BeerImage from "../../../components/beers/BeerImage";
import {
  Thumbnail,
  ImageWrapper
} from "../../../components/common/styles/Image";

const Props = {
  beer: {}
};

const wrapper = shallow(<Beer {...Props} />);
describe("render component", () => {
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe("toggle image", () => {
  it("should expand the image after clicking on thumbnail", () => {
    wrapper
      .find(BeerImage)
      .dive()
      .find(Thumbnail)
      .simulate("click");

    expect(wrapper.state().showImage).toEqual(true);
  });

  it("should close the image after clicking on image wrapper", () => {
    wrapper
      .find(BeerImage)
      .dive()
      .find(ImageWrapper)
      .simulate("click");

    expect(wrapper.state().showImage).toEqual(false);
  });
});
