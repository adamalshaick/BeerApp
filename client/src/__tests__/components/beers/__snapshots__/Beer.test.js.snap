// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<styled.div
  className="row"
>
  <div
    className="col-8"
  >
    <strong>
      name:
    </strong>
     
    <br />
    <strong>
      type:
    </strong>
     
    <br />
    <strong>
      price:
    </strong>
     
  </div>
  <div
    className="col-4"
  >
    <BeerImage
      beer={Object {}}
      onClick={[Function]}
      showImage={false}
    />
  </div>
</styled.div>
`;
